libcnb/src/target.rs

```diff
@@ -0,0 +1,36 @@
+pub struct ContextTarget {
+    /// The name of the target operating system.
+    ///
+    /// The value should conform to [Go's `$GOOS`](https://golang.org/doc/install/source#environment), for example
+    /// `linux` or `windows`.
+    ///
+    /// CNB `lifecycle` sources this value from the build OCI image's [`os` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
+    pub os: String,
+    /// The name of the target CPU architecture.
+    ///
+    /// The value should conform to [Go's $GOARCH](https://golang.org/doc/install/source#environment), for example
+    /// `amd64` or `arm64`.
+    ///
+    /// CNB `lifecycle` sources this value from the build OCI image's [`architecture` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
+    /// ``
+    pub arch: String,
+    /// The variant of the specified CPU architecture.
+    ///
+    /// The value should conform to [OCI image spec platform variants](https://github.com/opencontainers/image-spec/blob/main/image-index.md#platform-variants), for example
+    /// `v7` or `v8`.
+    ///
+    /// CNB `lifecycle` sources this value from the build OCI image's [`variant` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
+    pub arch_variant: Option<String>,
+    /// The name of the operating system distribution. Should be empty for windows.
+    ///
+    /// For example: `ubuntu` or `arch`.
```

Comment:
(Optional) I found the usage of "arch" in this example confusing for a split second, since I read it as not an example, but referring the the `arch` field above. Whilst I then realised it meant the Arch distro, perhaps using another distro name here would be clearer?

Also, I wasn't sure what the correct capitalisation would be for these fields (sadly it's another thing the spec doesn't advise on - perhaps we should file this upstream?).

I presume lowercase is recommended, since I found lowercase for a slightly similar opencontainer field that some images already set:

```
$ docker inspect ubuntu | rg opencontainers
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.version": "22.04"
```

Plus the OS release metadata uses lowercase for the `ID` field for several OSes:

```
$ dr ubuntu sh -c "cat /etc/*-release | sort | grep -v https"
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
ID=ubuntu
ID_LIKE=debian
NAME="Ubuntu"
PRETTY_NAME="Ubuntu 22.04.3 LTS"
UBUNTU_CODENAME=jammy
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
VERSION_ID="22.04"
```

```
$ dr debian sh -c "cat /etc/*-release | sort | grep -v https"
ID=debian
NAME="Debian GNU/Linux"
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
VERSION_ID="12"
```

```
$ dr alpine sh -c "cat /etc/*-release | sort | grep -v https"
3.19.1
ID=alpine
NAME="Alpine Linux"
PRETTY_NAME="Alpine Linux v3.19"
VERSION_ID=3.19.1
```

Reply:
Switched to `alpine` in [c5cb54c](https://github.com/heroku/libcnb.rs/pull/773/commits/c5cb54ce9285983331e3afb08accbb45e970c542).